### PR TITLE
feat: add metrics collector to all commands

### PR DESCRIPTION
**Pull request by PatrickMenoti**
**WHAT**
- ask user if they allow metrics to be collected (default: yes). 
  - saving the answer in the settings.toml file, using the following pattern: (0 = not answered yet; 1 = authorized; 2 = not authorized);
- PersistentPostRunE added: 
  - runs after every command is completed and updates the following metrics (if the user allowed):
    - `total-commands-executed` (count of every command executed by the user);
    - `command-subcommand` , for e.g.: `list-domain` (count of executions per command);

Metrics are being saved in a map of ints. In the future, when more complex metrics are added, we might need to update this variable to a map of interfaces perhaps;

### CODE REVIEW

File: pkg/cmd/root/post_command.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	var data map[string]int
+
+	decoder := json.NewDecoder(file)
+	err = decoder.Decode(&data)
+	if err != nil && err != io.EOF {
+		return err
+	}
+
+	// If EOF is encountered or the file is empty, initialize data as an empty map
+	if data == nil {
```
Comment (maxwelbm) [inline pkg/cmd/root/post_command.go:54]: 🥇

File: pkg/cmd/root/pre_command.go [hunk 2]
```diff
@@ -260,3 +267,23 @@ func linuxUpdateMessage(f *cmdutil.Factory) error {
 
 	return nil
 }
+
+// 0 = authorization was not asked yet, 1 = accepted, 2 = denied
```
Comment (maxwelbm) [inline pkg/cmd/root/pre_command.go:271]: just to follow golang's doc convention
```suggestion
// checkAuthorizeMetricsCollection 0 = authorization was not asked yet, 1 = accepted, 2 = denied
```

File: utils/helpers.go [hunk 3]
```diff
@@ -584,14 +584,16 @@ func Concat(strs ...string) string {
 	return sb.String()
 }
 
-func Confirm(msg string) bool {
-	fmt.Printf("🤔 \x1b[32m%s (y/N) \x1b[0m", msg)
+func Confirm(msg string, defaultYes bool) bool {
```
Comment (maxwelbm) [inline utils/helpers.go:587]: Much better 😄

**Comment by maxwelbm** [comment]
The only suggestion I would have would be to create a metrics directory within the pkg to centralize all metrics in one place. and leave the root directory with just the logic and the call. whatever is just a personal taste

**Review by maxwelbm — APPROVED** [review]

**Comment by PatrickMenoti** [comment]
> The only suggestion I would have would be to create a metrics directory within the pkg to centralize all metrics in one place. and leave the root directory with just the logic and the call. whatever is just a personal taste

We can do that along with the new metrics, when we do it!

**Review by jotanarciso — APPROVED** [review]

**Pull request merged**